lib/cache/FactoryCacheService.ts

```diff
@@ -0,0 +1,34 @@
+import { CacheOptions } from '.';
+import { Cache } from './Cache';
+import { DEFAULT_EXPIRATION, DEFAULT_OPTIONS, DEFAULT_SIZE, DEFAULT_STORAGE } from './CacheOptions';
+import { ExpirationStrategy } from './ExpirationStrategy';
+import { MemoryStorage } from './MemoryStorage';
+import { StorageType } from './StorageType';
+
+export function factoryCacheService<K = any>(
```

Comment:
A more appropriate name for the file is `CacheFactory`, and then I'd name the exported function just `create`